SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -695,123 +695,96 @@ void CFlowOutput::SetRotatingFrameCoefficients(CConfig *config, CSolver *flow_so
 }
 
 
-void CFlowOutput::Add_CpInverseDesignOutput(CConfig *config){
+void CFlowOutput::Add_CpInverseDesignOutput(){
 
   AddHistoryOutput("INVERSE_DESIGN_PRESSURE", "Cp_Diff", ScreenOutputFormat::FIXED, "CP_DIFF", "Cp difference for inverse design");
 
 }
 
-void CFlowOutput::Set_CpInverseDesign(CSolver *solver, CGeometry *geometry, CConfig *config){
-
-  unsigned short iMarker, icommas, Boundary;
-  unsigned long iVertex, iPoint, (*Point2Vertex)[2], nPointLocal = 0, nPointGlobal = 0;
-  su2double XCoord, YCoord, ZCoord, Pressure, PressureCoeff = 0, Cp, CpTarget, *Normal = nullptr, Area, PressDiff = 0.0;
-  bool *PointInDomain;
-  string text_line, surfCp_filename;
-  ifstream Surface_file;
+void CFlowOutput::Set_CpInverseDesign(CSolver *solver, const CGeometry *geometry, const CConfig *config){
 
   /*--- Prepare to read the surface pressure files (CSV) ---*/
 
-  surfCp_filename = config->GetUnsteady_FileName("TargetCp", (int)curTimeIter, ".dat");
-
-  /*--- Read the surface pressure file ---*/
+  const auto surfCp_filename = config->GetUnsteady_FileName("TargetCp", curTimeIter, ".dat");
 
-  string::size_type position;
+  /*--- Read the surface pressure file, on the first inner iteration. ---*/
 
+  ifstream Surface_file;
   Surface_file.open(surfCp_filename);
 
-  if (!(Surface_file.fail())) {
-
-    nPointLocal = geometry->GetnPoint();
-    SU2_MPI::Allreduce(&nPointLocal, &nPointGlobal, 1, MPI_UNSIGNED_LONG, MPI_SUM, SU2_MPI::GetComm());
-
-    Point2Vertex = new unsigned long[nPointGlobal][2];
-    PointInDomain = new bool[nPointGlobal];
-
-    for (iPoint = 0; iPoint < nPointGlobal; iPoint ++)
-      PointInDomain[iPoint] = false;
-
-    for (iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++) {
-      Boundary   = config->GetMarker_All_KindBC(iMarker);
-
-      if (config->GetSolid_Wall(iMarker) || (Boundary == NEARFIELD_BOUNDARY)) {
-        for (iVertex = 0; iVertex < geometry->GetnVertex(iMarker); iVertex++) {
-
-          /*--- The Pressure file uses the global numbering ---*/
-
-          iPoint = geometry->nodes->GetGlobalIndex(geometry->vertex[iMarker][iVertex]->GetNode());
-
-          if (geometry->vertex[iMarker][iVertex]->GetNode() < geometry->GetnPointDomain()) {
-            Point2Vertex[iPoint][0] = iMarker;
-            Point2Vertex[iPoint][1] = iVertex;
-            PointInDomain[iPoint] = true;
-            solver->SetCPressureTarget(iMarker, iVertex, 0.0);
-          }
-
-        }
-      }
-    }
+  if (!Surface_file.good()) {
```

Comment:
Nice!

Reply:
Ah! Because this function is always called and the result is always available as one of the outputs :shrug: and so it cannot throw an error.